src/core/web_api/p2_api.js

```diff
@@ -65,11 +68,13 @@ class Auth0APIClient {
     // TODO: for passwordless only, try to clean in auth0.js
     // client._shouldRedirect = redirect || responseType === "code" || !!redirectUrl;
     const f = loginCallback(false, this.domain, cb);
-    const loginOptions = normalizeAuthParams({
-      ...options,
-      ...this.authOpt,
-      ...authParams
-    });
+    const loginOptions = trimAuthParams(
```

Comment:
The function wrap feels ... wrong. Can `trimAuthParams` be a part of `normalizeAuthParams`? What if we need to do another transform here? Maybe there should be a `prepareParams` or similar to handle everything?